### PR TITLE
Go-1.9 is now default for juju 2.3

### PR DESCRIPTION
**Pull request by nskaggs**
## Description of change\
Why is this change needed? This change allows us to build with go-1.9, which will be required for new changes going forward.

## QA steps

The merge jobs have been updated as part of the PR to build with go-1.9. The readme and other docs also now mention go-1.9. Once this PR passes, builds should be good to go.

### CODE REVIEW

**Comment by nskaggs** [comment]
@Veebers thoughts?

File: README.md [hunk 1]
```diff
@@ -16,17 +16,14 @@ Getting started
...
-    sudo apt-add-repository ppa:juju/stable
-    sudo apt-get update
-    sudo apt-get install juju
+    snap install juju --classic
     
 Installing Go
 --------------
 
-`Juju's` source code currently depends on Go 1.8. One of the easiest ways
+`Juju's` source code currently depends on Go 1.9. One of the easiest ways
```
[the remark this answers is not in the document]
Comment (axw) [reply README.md]: +1

File: Makefile [hunk 2]
```diff
@@ -17,7 +17,7 @@ else
 endif
 
 ifeq ($(shell uname -p | sed -r 's/.*(86|armel|armhf|aarch64|ppc64le|s390x).*/golang/'), golang)
-	GO_C = golang-1.8
+	GO_C = golang-1.9
```
Comment (axw) [inline Makefile]: all of those architectures are now supported by gc - do we need to keep the gccgo bits hanging around?

Comment (nskaggs) [reply Makefile]: Ahh, right you are. I think gccgo bits can all be removed. I'll do so.

**Comment by dshcherb** [comment]
Might want to mark https://bugs.launchpad.net/juju/+bug/1516676 as "fix committed" after this is done.

**Comment by anastasiamac** [comment]
@dshcherb, 
Thank you for linking the bug, I have updated it as well \o/

**Comment by nskaggs** [comment]
@axw The cert unit test is causing an error. Can you have a look and correct?

**Comment by axw** [comment]
> @axw The cert unit test is causing an error. Can you have a look and correct?

https://github.com/juju/juju/pull/7935

**Comment by axw** [comment]
!!bang!!

**Comment by nskaggs** [comment]
$$merge$$

**Comment by nskaggs** [comment]
$$merge$$

**Pull request merged**